conda_forge_tick/migrators/pip_check.py

```diff
@@ -0,0 +1,68 @@
+import typing
+from typing import Any
+from ruamel.yaml import YAML
+
+from conda_forge_tick.xonsh_utils import indir
+from conda_forge_tick.migrators.core import MiniMigrator
+
+if typing.TYPE_CHECKING:
+    from ..migrators_types import AttrsTypedDict
+
+
+def _has_python_in_host(host):
+    host_set = {r.split(" ")[0] for r in host}
+    return bool(host_set & set(["python"])), host_set
+
+
+def _adjust_test_dict(test):
+    if 'requires' in test:
+        test['requires'].append('pip')
+    else:
+        test['requires'] = ['pip']
+
+    if 'commands' in test:
+        test['commands'].append('{{ PYTHON }} -m pip check')
+    else:
+        test['commands'] = ['{{ PYTHON }} -m pip check']
+
+
+class PipCheckMigrator(MiniMigrator):
+    def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
+        """run pip check if we see python in any host sections"""
+        host_reqs_list = (
```

Comment:
You should use the `requirements` key in the attrs itself it already has all the things broken out into sets.

Reply:
Does this key handle outputs too? I don't actually know what is in the attrs object.

Reply:
Yes, it also handles strong exports

Reply:
https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/make_graph.py#L120

Reply:
Where can I find the code that gives the layout of this data?

I don't think I need strong exports for this migrator but that is awesome!

Reply:
That block should give you the layout, it is a dict of sets.